### DISCUSSION TITLE
Checkbox component does not complain with WAI ARIA

### DISCUSSION THREAD

**Discussion by FilipePfluck** [chakra-ui/ark#3015] [General]
### Description

Doesn't have aria-checked

### Link to Reproduction (or Detailed Explanation)

https://ark-ui.com/react/docs/components/checkbox

### Steps to Reproduce

inspect elements and look the checkbox html

### Ark UI Version

4.1.12

### Framework

- [ ] React
- [ ] Solid
- [ ] Vue

**Comment by cschroeter** [comment]
hey @FilipePfluck 

please check the `HiddenInput` of the Checkbox component and you will find the roles you are looking for.

**Reply by FilipePfluck** [reply]
nope. It's true that it implicitly has the role='checkbox', but it still lacks the aria-checked

**Reply by FilipePfluck** [reply]
![Screenshot from 2024-11-11 19-33-20](https://github.com/user-attachments/assets/3d4a4f31-e20a-491d-96d0-85dfaec45be0)

**Comment by FilipePfluck** [comment]
The switch component also lacks role="switch" and aria-checked

**Comment by segunadebayo** [comment]
Closing — the accessibility attributes (`aria-checked`, `role`, etc.) are on the `HiddenInput` element, which is a native checkbox input. This is by design — screen readers interact with the native input rather than the visible control div.

```tsx
<Checkbox.Root>
  <Checkbox.Control />
  <Checkbox.HiddenInput /> {/* aria-checked, role="checkbox" live here */}
  <Checkbox.Label>Accept terms</Checkbox.Label>
</Checkbox.Root>
```

The same applies to the Switch component — use `Switch.HiddenInput` for the native input with proper ARIA attributes.
